containers/unit_tests/Makefile

```diff
@@ -31,14 +31,24 @@ KOKKOS_CXXFLAGS += -I$(GTEST_PATH) -I${KOKKOS_PATH}/containers/unit_tests
 TEST_TARGETS =
 TARGETS =
 
+TESTS = Bitset DualView DynamicView DynViewAPI_generic DynViewAPI_rank12345 DynViewAPI_rank67 ErrorReporter OffsetView ScatterView StaticCrsGraph UnorderedMap Vector ViewCtorPropEmbeddedDim
+tmp := $(foreach device, $(KOKKOS_DEVICELIST), \
+  tmp2 := $(foreach test, $(TESTS), \
+    $(if $(filter Test$(device)_$(test).cpp, $(shell ls Test$(device)_$(test).cpp 2>/dev/null)),,\
+      $(shell echo "\#include<Test"$(device)"_Category.hpp>" > Test$(device)_$(test).cpp); \
+      $(shell echo "\#include<Test"$(test)".hpp>" >> Test$(device)_$(test).cpp); \
+     )\
+  ) \
+)
+
 ifeq ($(KOKKOS_INTERNAL_USE_CUDA), 1)
 	OBJ_CUDA = UnitTestMain.o gtest-all.o
-	OBJ_CUDA += TestCuda_BitSet.o
+	OBJ_CUDA += TestCuda_Bitset.o
 	OBJ_CUDA += TestCuda_DualView.o
 	OBJ_CUDA += TestCuda_DynamicView.o
-	OBJ_CUDA += TestCuda_DynRankViewAPI_generic.o
-	OBJ_CUDA += TestCuda_DynRankViewAPI_rank12345.o
-	OBJ_CUDA += TestCuda_DynRankViewAPI_rank67.o
+	OBJ_CUDA += TestCuda_DynViewAPI_generic.o
```

Comment:
I don't know enough bash foo to see how to fix it, but something just feels wrong when we generate source files in a nice, clean loop above... then explicitly add each object file here.

Reply:
I can't figure it out. Inside a foreach loop I can apparently not append variables.